### PR TITLE
Update `RELEASING.md` to obtain steps to verify the changes for contrib

### LINKED ISSUE #5257: Add concrete steps to verify the code changes for contrib
### Problem Statement

There is a step `Update contrib codebase to support changes about to be released (use a git sha version)` in the release template issue, but the `RELEASING.md` didn't mention how actually do about it.

### Proposed Solution

- Add a makefile method in contrib to update otel dependencies to a certain git sha version.
- Add steps in the `RELEASING.md` file to verify the code changes can still support and contribute.

### Tasks

- [x] https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5434
- [ ] #5284

### PR DESCRIPTION
**Pull request by XSAM**
resolves #5257

### CODE REVIEW

**Review by dmathieu — APPROVED** [review]

File: RELEASING.md [hunk 1]
```diff
@@ -27,6 +27,37 @@ You can run `make gorelease` that runs [gorelease](https://pkg.go.dev/golang.org
 
 You can check/report problems with `gorelease` [here](https://golang.org/issues/26420).
 
+## Verify changes for contrib repository
+
+If the changes in the main repository are going to affect the contrib repository, it is important to verify that the changes are compatible with the contrib repository.
+
+Follow the following steps in the local [contrib repository](https://github.com/open-telemetry/opentelemetry-go-contrib) to verify the changes.
+
+1. Pick the GIT SHA on the [main branch](https://github.com/open-telemetry/opentelemetry-go/commits/main) that you want to verify.
```
Comment (MrAlias) [inline RELEASING.md:36]: These seem like steps relevant to contrib. Should they live in the contrib `RELEASING.md` instead?

Comment (XSAM) [reply RELEASING.md:36]: I can move this part to contrib and make a reference in this section.

Comment (XSAM) [reply RELEASING.md:36]: Blocked by https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5527

File: RELEASING.md [hunk 2]
```diff
@@ -27,6 +27,12 @@ You can run `make gorelease` that runs [gorelease](https://pkg.go.dev/golang.org
 
 You can check/report problems with `gorelease` [here](https://golang.org/issues/26420).
 
+## Verify changes for contrib repository
+
+If the changes in the main repository are going to affect the contrib repository, it is important to verify that the changes are compatible with the contrib repository.
+
+Follow [the steps](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/RELEASING.md#verify-otel-changes) in the contrib repostory to verify OTel changes.
```
Comment (MrAlias) [inline RELEASING.md:34]:
```suggestion
Follow [the steps](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/RELEASING.md#verify-otel-changes) in the contrib repository to verify OTel changes.
```

**Review by MrAlias — APPROVED** [review]

**Pull request merged**